lib/utils/resource-links.js

```diff
@@ -0,0 +1,60 @@
+function isResourceLink (field) {
+  return field && field.sys && field.sys.type === 'ResourceLink'
+}
+
+function getResourceLinkFields (data) {
+  const items = [].concat(...(data.items || []), ...Object.values(data.includes || {}).flat())
+  const fields = items.flatMap(({ fields }) => Object.values(fields).flat())
+  return fields.filter(isResourceLink)
+}
+
+function parseResourceLink (resourceLink) {
+  const urn = resourceLink.sys.urn
+  // For now just assume every resourceLink has type Contentful:Entry
+  const [, spaceId, , entryId] = urn.split('/')
+  return { spaceId, entryId, urn }
+}
+
+function groupBy (key, objArray) {
+  return objArray.reduce((acc, obj) => {
+    const objKey = obj[key]
+    acc[objKey] = (acc[objKey] || []).concat(obj)
+    return acc
+  }, {})
+}
+
+async function getEntitiesFromSpace (spaceId, entries, extraApis, include) {
+  const apiInstance = extraApis[spaceId]
+  if (!apiInstance) {
+    return []
+  }
+
+  const entryIds = entries.map(({ entryId }) => entryId)
+
+  let response = { items: [] }
+  const uniqueEntryIds = [...new Set(entryIds)]
+  response = await apiInstance.getEntries({ 'sys.id[in]': uniqueEntryIds.join(','), include }).catch(() => ({}))
+  return response.items || []
+}
+
+export async function addResourceLinksToIncludes (data, extraApis = {}, include = 1) {
```

Comment:
FYI: This function is not written recursively. You'll only be able to resolve 1 layer of x-space ResourceLinks.

Reply:
It is, `addResourceLinksToIncludes` calls `getEntitiesFromSpace` which calls `getEntries` which calls `addResourceLinksToIncludes` again

Reply:
You are right. I made the same mistake in my other comment where I forgot that every api instance is doing its own recursive resolving

Reply:
I added a test case for it